tests/transport/compliance.py

```diff
@@ -117,27 +126,31 @@ def test_connection_error(self):
         request = self.make_request()
         with pytest.raises(exceptions.TransportError):
             request(url="http://{}".format(NXDOMAIN), method="GET")
-        
-    def test_cached_request(self, server):
-        request = self.make_cached_request()
-        response = request(url=server.url + "/cache", method="GET")
-        request_time = response.headers.get("Time")
-        
-        time.sleep(_SLEEP_TIME)
-        
-        new_response = request(url=server.url + "/cache", method="GET")
-        new_time = new_response.headers.get("Time")
-        
-        assert request_time == new_time, "{} and {} are not equal".format(request_time, new_time)
+
+    def test_cached_request(self, server, monkeypatch):
+        with freeze_time("2000-01-01 00:00:00") as frozen_datetime:
+            request = self.make_cached_request()
+            response = request(url=server.url + "/cache", method="GET")
+            request_time = response.headers.get("Time")
+            
+            monkeypatch.setattr(time, 'sleep', sleepless)
```

Comment:
Instead of mocking ```time.sleep```, it would be simpler to just directly call ```sleepless``` and maybe rename it to ```set_time```